src/components/HighlightedLink.js

```diff
@@ -1,6 +1,43 @@
+import PropTypes from "prop-types";
 import { Link } from "gatsby";
+import { useLocation } from "react-use";
 import styled from "styled-components";
 
-export default styled(Link).attrs({
+const StyledLink = styled(Link).attrs({
   className: "text-primary underline-offset-2 decoration-1 decoration-dotted hover:text-primary-light hover:underline",
 })``;
+
+/**
+ * This an extended version of Gatsby's <Link /> component, which
+ * is also styled to match Skynet Labs' dashboard's theme.
+ */
+export default function HighlightedLink({ persistQueryString, to, ...props }) {
+  const { search, origin } = useLocation();
+  let url = to;
+
+  if (persistQueryString) {
+    const targetUrl = new URL(url, origin); // Use current origin as a base (it will be discarded it `url` is an absolute URL)
+    const currentQueryParams = new URLSearchParams(search);
+    currentQueryParams.forEach((value, key) => {
+      // Do not override query params if they exist already
+      if (!targetUrl.searchParams.has(key)) {
+        targetUrl.searchParams.set(key, value);
+      }
+    });
+
+    url = targetUrl.toString().replace(origin, "");
+  }
```

Comment:
This works now but I think we will need to refactor that soon enough anyway since we are going to have a flow where we require user to sign up and then pay and then only after we're back from stripe and payment went through, redirected to the skylink.